keylime/ima/ima.py

```diff
@@ -287,6 +288,7 @@ def _process_measurement_list(
     if not is_valid:
         # This should not happen as the exclude list has already been validated
         # by the verifier before acceping it. This is a safety net just in case.
+        assert err_msg is not None
```

Comment:
I now put in PR #1185 to address this particular issue that we have here due to redundancy and we can avoid the assert if we want to.

Reply:
Now that PR #1185 has been merged you should be able to remove this assert. 

The issue with the asserts is that that they may in some cases hide issues that we can fix but in other cases pyright simply doesn't understand more complex code and it may be necessary to add them then.